### PR TITLE
feat: Trino blueprint upgrade

### PR DESCRIPTION
**Pull request by vara-bonthu**
### What does this PR do?

🛑 Please open an issue first to discuss any significant work and flesh out details/direction. When we triage the issues, we will add labels to the issue like "Enhancement", "Bug" which should indicate to you that this issue can be worked on and we are looking forward to your PR. We would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/awslabs/data-on-eks/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

- Trino Blueprint upgrade with latest addons
- Trino helm chart config to scale for large scale workloads

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Mandatory for new blueprints. Yes, I have added a example to support my blueprint PR
- [ ] Mandatory for new blueprints. Yes, I have updated the `website/docs` or `website/blog` section for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR. Link for installing [pre-commit](https://pre-commit.com/) locally

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: distributed-databases/trino/helm-values/trino.yaml [hunk 1]
```diff
@@ -69,17 +145,4 @@ serviceAccount:
   create: true
   name: ${sa}
 ingress:
-  enabled: true
```
Comment (vara-bonthu) [inline distributed-databases/trino/helm-values/trino.yaml:72]: Removed to avoid creating a public load balancer

**Review by ratnopamc — APPROVED** [review]
lgtm

File: distributed-databases/trino/helm-values/trino.yaml [hunk 2]
```diff
@@ -7,44 +19,108 @@ server:
...
+      maxExecutionTime: "24h"
+      clientTimeout: "2h"
 service:
   type: ClusterIP
   port: 8080
+
 coordinator:
   jvm:
-    maxHeapSize: "8G"
+    maxHeapSize: "32G"  # ~80% of container memory
```
[the remark this answers is not in the document]
Comment (vara-bonthu) [reply distributed-databases/trino/helm-values/trino.yaml:38]: Agree! Planning to update the Trino website doc in the second PR with all the best practices.

File: distributed-databases/trino/helm-values/trino.yaml [hunk 3]
```diff
@@ -7,44 +19,108 @@ server:
...
+
 additionalConfigProperties:
   - "retry-policy=TASK"
   - "exchange.compression-enabled=true"
-  - "query.low-memory-killer.delay=0s"
   - "query.remote-task.max-error-duration=1m"
-  - "query.hash-partition-count=50"
+  - "query.max-hash-partition-count=100"   # Updated from query.hash-partition-count
+  - "spill-enabled=true"                   # Updated from experimental.spill-enabled
+  - "spiller-spill-path=/tmp/spill"        # Chagne this to SSD mount for faster
```
Comment (ratnopamc) [inline distributed-databases/trino/helm-values/trino.yaml:109]: Can we add note in the documentation about spill to disk and how it helps.

**Pull request merged**